Userland/useradd.cpp

```diff
@@ -0,0 +1,104 @@
+#include <AK/String.h>
+#include <LibCore/CArgsParser.h>
+#include <pwd.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+static constexpr uid_t BASE_UID = 1000;
+static constexpr gid_t USERS_GID = 100;
+const char* DEFAULT_SHELL = "/bin/sh";
+
+int main(int argc, char** argv)
+{
+    CArgsParser args_parser("useradd");
+    args_parser.add_arg("d", "HOME_DIR", "home directory for the new user");
+    args_parser.add_arg("u", "UID", "user ID (uid) for the new user");
+    args_parser.add_arg("g", "GROUP", "group ID (gid) for the new user");
+    args_parser.add_arg("m", "create home directory if it does not exist");
+    args_parser.add_arg("s", "SHELL", "name of default shell binary for the new user");
+    args_parser.add_arg("n", "NAME", "gecos name of the new user");
+    CArgsParserResult args = args_parser.parse(argc, argv);
+
+    if (argc <= 1) {
+        args_parser.print_usage();
+        return -1;
+    }
+
+    const char* username = argv[argc - 1]; // Always the user name!
+    uid_t uid = BASE_UID;
+    gid_t gid = USERS_GID;
+
+    // First, let's sort out the uid for the user
+    if (args.is_present("u")) {
+        bool ok;
+        uid = static_cast<uid_t>(args.get("u").to_uint(ok));
```

Comment:
What if this `UID` already exists?

Reply:
Forgot about that. According to `man useradd` on Linux, I should be returning a value of `4`.